packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1707,10 +1707,14 @@ class _Decorator extends RenderObjectWidget {
   void updateRenderObject(BuildContext context, _RenderDecoration renderObject) {
     renderObject
      ..decoration = decoration
-     ..textDirection = textDirection
-     ..textBaseline = textBaseline
      ..expands = expands
-     ..isFocused = isFocused;
+     ..isFocused = isFocused
+     ..textBaseline = textBaseline
+     ..textDirection = textDirection;
+
+    if (textAlignVertical != null) {
```

Comment:
There is an assert to make sure that _RenderDecoration.textAlignVertical is not set to null, so I had to use this `if`.

Reply:
Nit: it doesn't do anything if `textAlignVertical` is set to null from a non-null value? That sounds a little weird. `textAlignVertical` seems to be nullable but I couldn't find the corresponding behavior in the public documentation. Maybe we should document it?